delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/client/DeltaClient.scala

```diff
@@ -1,44 +1,99 @@
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.persistence.query.{NoOffset, Offset, Sequence, TimeBasedUUID}
+import akka.stream.alpakka.sse.scaladsl.EventSource
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.client.DeltaClient._
 import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewSource.RemoteProjectSource
 import ch.epfl.bluebrain.nexus.delta.rdf.RdfMediaTypes
-import ch.epfl.bluebrain.nexus.delta.sdk.http.{HttpClient, HttpClientError}
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClient
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClient.HttpResult
 import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.AuthToken
 import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectCountsCollection.ProjectCount
 import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
-import monix.bio.IO
+import com.typesafe.scalalogging.Logger
+import io.circe.Decoder
+import io.circe.parser.decode
+import fs2._
+import monix.bio.{IO, Task}
+import monix.execution.Scheduler
+import streamz.converter._
+
+import java.util.UUID
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Try
 
 /**
   * A client to request a remote delta instance
   */
-final class DeltaClient(client: HttpClient) {
+final class DeltaClient(client: HttpClient, retryDelay: FiniteDuration)(implicit
+    as: ActorSystem[Nothing],
+    scheduler: Scheduler
+) {
 
   /**
     * Fetches statistics for the remote source
     */
-  def statistics(source: RemoteProjectSource): IO[HttpClientError, ProjectCount] = {
-    implicit val cred: Option[AuthToken] = source.token.map { token => AuthToken(token.value.value) }
+  def statistics(source: RemoteProjectSource): HttpResult[ProjectCount] = {
+    implicit val cred: Option[AuthToken] = token(source)
     val statisticsEndpoint: HttpRequest  =
       Get(
         source.endpoint / "projects" / source.project.organization.value / source.project.project.value / "statistics"
       ).addHeader(accept).withCredentials
     client.fromJsonTo[ProjectCount](statisticsEndpoint)
   }
 
+  private def toOffset(id: String): Offset =
+    Try(TimeBasedUUID(UUID.fromString(id))).orElse(Try(Sequence(id.toLong))).getOrElse(NoOffset)
+
+  def events[A: Decoder](source: RemoteProjectSource, offset: Offset): Stream[Task, (Offset, A)] = {
+    val initialLastEventId = offset match {
+      case NoOffset             => None
+      case Sequence(value)      => Some(value.toString)
+      case TimeBasedUUID(value) => Some(value.toString)
+    }
+
+    implicit val cred: Option[AuthToken] = token(source)
+
+    def send(request: HttpRequest): Future[HttpResponse] = {
+      client[HttpResponse](request.withCredentials)(IO.pure(_)).runToFuture
+    }
+
+    val uri =
+      source.endpoint / "resources" / source.project.organization.value / source.project.project.value / "events"
```

Comment:
will this return all the events for a project (incl: resolvers, schemas etc.)?

Reply:
yes, this endpoint currently returns aggregated events